EIPS/eip-4361-1.md

```diff
@@ -0,0 +1,238 @@
+
+SIWE CapGroks unlock integration of (decentralized) protocols and/or APIs for developers by reducing user friction, onchain state and increasing security by introducing informed consent and deterministic capability objects.
+
+This specification has three different audiences:
+- Web3 application developers that want to integrate CapGroks to interact with Web3 protocols and APIs.
+- Web3 protocol or API developers that want to learn how to define their own CapGroks.
+- Wallet implementers that want to improve the UI for capgroks.
+
+## Terms and Definitions
+
+(non-normative)
+
+- CapGrok - A SIWE Message complying with this specification, i.e., containing at least one CapGrok URI in the `Resources` section and the corresponding human-readable CapGrok Statement appended to the SIWE `statement`.
+- CapGrok URI - A type of URI under a certain namespace that resolves to a CapGrok Details Object.
+- CapGrok Details Object - A JSON object describing the actions and optionally the resources associated with a CapGrok Capability under a certain namespace.
+- Resource Service (RS) - The entity that is providing third-party services for the Ethereum account.
+- SIWE Client (SC) - The entity initiating the SIWE authentication and CapGrok flow.
+- Relying Party (RP) - same as SC in the context of authentication.
+
+## Specification
+
+(normative)
+
+### Overview
+
+This specification defines the following:
+- CapGrok SIWE Extension
+- CapGrok Capability
+   - CapGrok URI Scheme
+   - CapGrok Details Object Schema
+- CapGrok Translation Algorithm
+- CapGrok Verification
+- CapGrok Namespace Registry
+
+### CapGrok SIWE Extension
+
+A CapGrok is a EIP-4361 message following a specific format that allows an Ethereum account to delegate a set of CapGrok Capabilities to a delegee through informed consent. Each CapGrok Capability MUST be represented by an entry in the `Resources` array of the SIWE message that MUST deterministically translate the CapGrok Capability in human-readable form to the `statement` field in the SIWE message using the CapGrok Translation Algorithm.
+
+The following SIWE message fields are used to further define (or limit) the scope of all CapGrok Capabilities:
+- The `URI` field MUST specify the delegee as a URI, e.g., `https://example.com`, `did:key:z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK`. It is expected that the RS authenticates the delegee before invoking an action for the CapGrok Capability.
+- The `Issued At` field MUST be used to specify the issuance date of the CapGrok Capabilities.
+- If present, the `Expiration Time` field MUST be used as the expiration time of the CapGrok Capabilities where the RS no longer accepts invoking the associated actions.
+- If present, the `Not Before` field MUST be used as the time that has to expire before the RS starts accepting invoking of the associated actions of the CapGrok Capabilities.
```

Comment:
These fields are optional in the SIWE message usually. To be clear, do we want to enforce these fields to be _required_ for a CapGrok SIWE?

Reply:
nope, that is why there is this `If present` language. i can make it more clear.

Reply:
No sorry that is clear, I just skim-read it and miss the `If present`